### PR TITLE
Add docs for jest-emotion toHaveStyleRule additional options

### PR DESCRIPTION
**Pull request by Werter12**
I've noticed that some functionality of `toHaveStyleRule` matcher is not clear for developers https://github.com/emotion-js/emotion/issues/1208 . So this PR intended to tell about additional options which developer can provide for `toHaveStyleRule` matcher like `target` and `media`.
- [x] Documentation

### CODE REVIEW

File: packages/jest-emotion/README.md [hunk 1]
```diff
@@ -94,14 +94,67 @@ import styled from '@emotion/styled'
...
+    .toJSON()
 
   expect(tree).toHaveStyleRule('float', 'left')
-  expect(tree).not.toHaveStyleRule('color', 'hotpink')
+  expect(tree).not.toHaveStyleRule('height', '100%')
+})
+```
+
+You can provide additional options for `toHaveStyleRule` matcher.  
+`target` - option helps to specify target where style rule should be found.
```
Comment (mitchellhamilton) [inline packages/jest-emotion/README.md]: This should mention that's it's a css selector.

File: packages/jest-emotion/README.md [hunk 2]
```diff
@@ -94,14 +94,67 @@ import styled from '@emotion/styled'
...
+
+1.  Pseudo-class.
```
Comment (mitchellhamilton) [inline packages/jest-emotion/README.md]: Could you remove the 1., 2. and etc.?

File: packages/jest-emotion/README.md [hunk 3]
```diff
@@ -94,14 +94,67 @@ import styled from '@emotion/styled'
...
+expect(tree).toHaveStyleRule('color', 'yellow', { target: 'span' })
+```
+
+3. Another styled component.
+
+```
+expect(tree).toHaveStyleRule('fill', 'green', { target: `${Svg}` })
+```
+
+`media` - option allows to test style rule within media.
```
Comment (mitchellhamilton) [inline packages/jest-emotion/README.md]: This doesn't seem like a great description of what it does, something along the lines of 'specifies the media rule that the matcher should look for the property in'. or that worded better

File: packages/jest-emotion/README.md (hunk 1 above)
Comment (Werter12) [reply packages/jest-emotion/README.md]: yes

File: packages/jest-emotion/README.md (hunk 2 above)
Comment (Werter12) [reply packages/jest-emotion/README.md]: sure

File: packages/jest-emotion/README.md (hunk 3 above)
Comment (Werter12) [reply packages/jest-emotion/README.md]: Ok

**Pull request merged**